api/v1beta1/externaldns_types.go

```diff
@@ -101,6 +101,11 @@ type ExternalDNSSpec struct {
 	// +kubebuilder:validation:Optional
 	// +optional
 	Zones []string `json:"zones,omitempty"`
+
+	// AWSRoleARN is a string
+	// +kubebuilder:validation:Optional
+	// +optional
+	AWSRoleARN *string `json:"awsRoleARN,omitempty"`
```

Comment:
`RoleARN` is better to be put into [AWSproviderOptions](https://github.com/openshift/external-dns-operator/blob/0cd8bee31e64f3086f9a44e3f5e299a3a1111175/api/v1beta1/externaldns_types.go#L246) as it's AWS specific.

Reply:
Yea this was intentionally sloppy to get a POC, but I didn't notice the `provider` field, that makes much more sense, will do soon.